hubspot3/tickets.py

```diff
@@ -168,7 +176,7 @@ def get_all_as_generator(self, limit: int = -1, extra_properties: Union[List[str
                     **options
                 )
                 unjoined_outputs.extend(batch["objects"])
-            outputs = self._join_output_properties(unjoined_outputs)
+            outputs = self._join_output_properties_to_list(unjoined_outputs)
```

Comment:
This seems to be the only call to this function. I don't think it is worth having it as a separate function.

Reply:
I think it does, it makes the code clearer and easier to read. And also improve the separation of responsibility

Reply:
Requiring reading two methods is a readability cost too. And in this case the other method simply does `list(x.values()`.

```python
outputs = self._join_output_properties(unjoined_outputs)
output_values = list(outputs.values())
```

The name of the variable is enough to tell me what is happening, the `_to_list` doesn't mean much, and even worse it is implying the whole thing is converted to a list, when only the values are.

If we are going to have two methods, at least let's name it properly and clearly define what it is supposed to do.

Reply:
When we discussed this, we realized that the only difference between the functions was the conversion for list. We chose breaking in two functions over converting the output outside to keep the code more semantic.